subaccounts.go

```diff
@@ -44,30 +49,30 @@ func (c *Client) SubaccountCreate(s *Subaccount) (res *Response, err error) {
 func (c *Client) SubaccountCreateContext(ctx context.Context, s *Subaccount) (res *Response, err error) {
 	// enforce required parameters
 	if s == nil {
-		err = fmt.Errorf("Create called with nil Subaccount")
+		err = errors.New("Create called with nil Subaccount")
 	} else if s.Name == "" {
-		err = fmt.Errorf("Subaccount requires a non-empty Name")
+		err = errors.New("Subaccount requires a non-empty Name")
 	} else if s.KeyLabel == "" {
-		err = fmt.Errorf("Subaccount requires a non-empty Key Label")
+		err = errors.New("Subaccount requires a non-empty Key Label")
 	} else
 	// enforce max lengths
 	if len(s.Name) > 1024 {
-		err = fmt.Errorf("Subaccount name may not be longer than 1024 bytes")
+		err = errors.New("Subaccount name may not be longer than 1024 bytes")
 	} else if len(s.KeyLabel) > 1024 {
-		err = fmt.Errorf("Subaccount key label may not be longer than 1024 bytes")
+		err = errors.New("Subaccount key label may not be longer than 1024 bytes")
+	} else if s.IPPool != "" && len(s.IPPool) > 20 {
+		err = errors.New("Subaccount ip pool may not be longer than 20 bytes")
 	}
 	if err != nil {
 		return
 	}
 
 	if len(s.Grants) == 0 {
-		s.Grants = availableGrants
+		s.Grants = SubaccountGrants
 	}
 
-	jsonBytes, err := json.Marshal(s)
-	if err != nil {
-		return
-	}
+	// Marshaling a static type won't fail
+	jsonBytes, _ := json.Marshal(s)
 
 	path := fmt.Sprintf(SubaccountsPathFormat, c.Config.ApiVersion)
```

Comment:
I'd rather have the option to easily support multiple versions in case that's needed.

That said, to simplify the repeated code, the `fmt.Sprintf` could perhaps be moved into `Client.Init` to populate something like `Client.PathPrefix` which could then get joined together with the other components more simply.

Please open a separate issue for that, and I can see what makes sense.